syntax = "proto2";

package mipt_distencode;

import "mipt_distencode/jobs.proto";
import "mipt_distencode/mgmt_messages.proto";

service Worker {
    rpc GetState(WorkerIdentity) returns (WorkerSelfAnnouncement) {}

    rpc Ping(PingMesg) returns (PingMesg) {}

    // Expects that job.id is set
    // Returns the same job id
    rpc PostMeltJob(MeltJob) returns (JobId) {}
}
